SYCL/Basic/half_builtins.cpp

```diff
@@ -0,0 +1,227 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %HOST_RUN_PLACEHOLDER %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// OpenCL CPU driver does not support cl_khr_fp16 extension
+// UNSUPPORTED: cpu && opencl
+
+#include <CL/sycl.hpp>
+
+#include <cmath>
+#include <unordered_set>

```

Comment:
```suggestion
#include <limits>
```
This test doesn't use `std::unordered_set`.